CRM/Admin/Form/PaymentProcessor.php

```diff
@@ -332,9 +342,13 @@ public function setDefaultValues() {
 
     CRM_Core_DAO::storeValues($dao, $defaults);
     // When user changes payment processor type, it is passed in via $this->_ppType so update defaults array.
-    if ($this->_ppType) {
+    if ($this->_ppType && array_key_exists($this->_ppType, $this->_ppTypes)) {
```

Comment:
putting ppTypes in a static var is not actually gaining any efficiency over calling CRM_Core_PseudoConstant::paymentProcessorType() since it is already cached